packages/trader/src/App/Components/Form/RangeSlider/range-slider.jsx

```diff
@@ -6,82 +6,128 @@ import { localize } from 'App/i18n';
-                    max_value={max_value}
-                    name={name}
-                    steps={max_value - min_value}
-                    onChange={handleChange}
-                    tabIndex='0'
-                    value={value}
-                />
-                <div className='range-slider__ticks'>
-                    <TickSteps
+    resetOnHover = () => {
+        if (this.state.hover_value) {
+            this.setState({ hover_value: 0 });
+            this.rangeSliderTrack.style.width = 0;
+        }
+    };
+
+    render() {
+        const {
+            className,
+            max_value,
+            min_value,
+            name,
+            value,
+            onChange,
+        } = this.props;
+
+        if (+value < min_value || +value > max_value) {
+            onChange({
+                target: {
+                    name,
+                    value: min_value,
+                },
+            });
+        }
+
+        const display_value = this.state.hover_value || value;
+        return (
+            <div className={classNames('range-slider', className, { 'range-slider__error': ((value < +min_value) || (value > +max_value)) })}>
+                <label className='range-slider__label' htmlFor='range'>
+                    <input
+                        id='range'
+                        className='input trade-container__input range-slider__track'
+                        type='range'
+                        min={min_value}
+                        max={max_value}
+                        min_value={min_value}
+                        max_value={max_value}
+                        name={name}
+                        steps={max_value - min_value}
+                        onChange={this.handleChange}
+                        tabIndex='0'
                         value={value}
-                        onClick={handleClick}
                     />
-                </div>
-                {/* Calculate line width based on active value and size of range thumb */}
-                <div
-                    className='range-slider__line'
-                    style={{ width: `calc(${(value - min_value) * (100 / (max_value - min_value))}% ` }}
-                />
-            </label>
-            <div className='range-slider__caption'>
-                <span className='range-slider__caption--first'>
-                    {min_value}
-                </span>
-                {
-                    !!value &&
-                    <span className='range-slider__caption-title'>
-                        {+value === 1 && localize('{{value}} Tick', { value })}
-                        {+value > 1 && localize('{{value}} Ticks', { value })}
+                    {/* this element to be placed before <TickSteps /> to prevent overlapping colors */}
+                    <span
+                        ref={node => this.rangeSliderTrack = node}
+                        className='range-slider__line range-slider__line--hover'
```

Comment:
Maybe a different modifier name instead of `hover` on `onMouseEnter`? Since `span` element always going to be on `hover` upon render. Maybe `--fill` instead?